.circleci/config.yml

```diff
@@ -13,12 +13,15 @@ executors:
 aliases:
   - &restore_cache
     restore_cache:
-      key: dependencies-{{ .Branch }}-{{ checksum "package-lock.json" }}-7
+      key: dependencies-{{ .Branch }}-{{ checksum "package-lock.json" }}-8
   - &save_cache
     save_cache:
-      key: dependencies-{{ .Branch }}-{{ checksum "package-lock.json" }}-7
+      key: dependencies-{{ .Branch }}-{{ checksum "package-lock.json" }}-8
       paths:
         - ~/mattermost/mattermost-webapp/node_modules
+        - ~/mattermost/mattermost-webapp/packages/client/lib
```

Comment:
Our CircleCI config caches node_modules between steps so that it doesn't need to repeatedly re-download those for separate steps like the style check and type checks. NPM sets up symlinks for these internal-but-still-an-external-package dependencies from node_modules to the packages folder, but the directories linked to aren't cached as part of caching node_modules.

I was hoping to find a way to do this automatically for all packages, but I couldn't find a good way to do that. I added a note to that packages README to add this here as well.

Also, I don't think this is needed for GitLab CI since it only runs the one job.